server/src/test/java/io/druid/server/DruidNodeTest.java

```diff
@@ -32,7 +33,8 @@ public void testDefaultsAndSanity() throws Exception
     node = new DruidNode(service, null, null);
     Assert.assertEquals(DruidNode.getDefaultHost(), node.getHost());
     Assert.assertEquals(-1, node.getPort());
-    Assert.assertEquals(DruidNode.getDefaultHost(), node.getHostAndPort());
+    // Hosts which report an ipv6 address need a little extra conversion here
+    Assert.assertEquals(HostAndPort.fromString(DruidNode.getDefaultHost()).toString(), node.getHostAndPort());
```

Comment:
`getDefaultHost()` reports something like `fe80::6e40:8ff:fe93:9230` but `getHostAndPort()` reports `[fe80::6e40:8ff:fe93:9230]`

Rather than jump through hoops to determine what kind of address getDefaultHost is reporting, this was a much more straightforward approach.